### ISSUE TITLE
Ability to create custom VPC much like KOPS does

### ISSUE THREAD

**Issue by nukepuppy** [weaveworks/eksctl#279]
**Why do you want this feature?**
It would be wise to use one and not many (other tools) to provision an EKS cluster.  While using a tool like kops / terraform one can define things such as the CIDR block.  

Ability to set the CIDR block to  custom range, and define the subnet CIDR ranges would allow for organizations with pre-existing IP space to provision appropriate CIDRs for VPCs which can then be peered.

**What feature/behavior/change do you want?**
EKSctl to allow settings a --vpc-cidr  and then subnet-cidrs

There is another issue here: https://github.com/weaveworks/eksctl/issues/42  where I pointed out this issue and seemed that it should of be a "new issue"

values from something like kops look like
```
  networkCIDR: 172.31.0.0/16
...
  subnets:
  - cidr: 172.31.80.0/20
    id: subnet-5cae7172
    name: us-east-1a
    type: Public
    zone: us-east-1a
  - cidr: 172.31.16.0/20
    id: subnet-e9b2d3a3
    name: us-east-1b
    type: Public
    zone: us-east-1b
```

**Comment by errordeveloper** [comment]
As discuss on slack, we'd be happy to provide more configuration options, but it'd be easier to do after we have proper Cluster API support (0.3.0). Until then the things that can be done easily enough would be the following:

- `--vpc-cidr` global CIDR that will be subdivided as `eksctl` sees fit for the required number of AZs
- per-nodegroup control of public/private routing (after #257)

**Comment by rainmanh** [comment]
I would be also interested on re-using VPC/Subnets I got instead of letting the tool to create all the resources.
On some scenarios we need to re-use Network infrastructure instead of creating it. Sometimes we are limited and We cannot create it new.

**Comment by errordeveloper** [comment]
@rainmanh there is a general issue that is still open (#42), but we have already added support for re-using a VPC where kops cluster resides (which is mostly for migration and cluster peering use-cases). We are getting close to being able to cater for fixing #42 also.
